### PR TITLE
Allow users enter their card expiry date easily; provide better UX

### PR DESCRIPTION
**Pull request by emmanuelkehinde**
Considering what most users have gotten used to, allowing them to enter their expiry date in text will provide a better user experience.

### CODE REVIEW

**Comment by biodunalfet** [comment]
Looks good. Will test and merge soon. Thanks

File: raveandroid/src/main/java/com/flutterwave/raveandroid/card/CardFragment.java [hunk 1]
```diff
@@ -849,4 +832,58 @@ public void onPageFinished(WebView view, String url) {
...
+        @Override
+        public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+
+        }
+
+        @Override
+        public void afterTextChanged(Editable editable) {
+            String input = editable.toString();
+            SimpleDateFormat simpleDateFormat = new SimpleDateFormat("MM/yy");
+            Calendar calendar = Calendar.getInstance();
```
Comment (IEnoobong) [inline raveandroid/src/main/java/com/flutterwave/raveandroid/card/CardFragment.java]: Nice one man, learnt something new :smile: 

You should consider  creating a static `Calendar` as  cost of calling `Calendar.getInstance` every time text is changed is quite high [see](http://cephas.net/blog/2006/02/25/the-cost-of-calendar-object-creation/) [see](http://www.informit.com/articles/article.aspx?p=1216151&seqNum=5)

Comment (emmanuelkehinde) [reply raveandroid/src/main/java/com/flutterwave/raveandroid/card/CardFragment.java]: Oh...true.

I will fix that. Thanks

Comment (emmanuelkehinde) [reply raveandroid/src/main/java/com/flutterwave/raveandroid/card/CardFragment.java]: It's updated now.

File: raveandroid/src/main/java/com/flutterwave/raveandroid/card/CardFragment.java [hunk 2]
```diff
@@ -850,7 +855,6 @@ public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
         public void afterTextChanged(Editable editable) {
             String input = editable.toString();
             SimpleDateFormat simpleDateFormat = new SimpleDateFormat("MM/yy");
```
Comment (biodunalfet) [inline raveandroid/src/main/java/com/flutterwave/raveandroid/card/CardFragment.java]: You're creating an instance of `simpleDateFormat` each time the text changes which is expensive. Try to create it once in the `ExpiryWatcher()`'s constructor and use that instance in `line 860` instead just like you did for your `calendar` variable.

Comment (emmanuelkehinde) [reply raveandroid/src/main/java/com/flutterwave/raveandroid/card/CardFragment.java]: Oh true. Correcting that...

Comment (emmanuelkehinde) [reply raveandroid/src/main/java/com/flutterwave/raveandroid/card/CardFragment.java]: DONE.

**Pull request merged**